src/library/app_builder.cpp

```diff
@@ -754,6 +754,54 @@ class app_builder {
         level lvl_2  = get_level(B);
         return ::lean::mk_app({mk_constant(get_congr_arg_name(), {lvl_1, lvl_2}), A, B, lhs, rhs, f, H});
     }
+
+    expr mk_congr_fun(expr const & H, expr const & a) {
+        expr eq = m_ctx.relaxed_whnf(m_ctx.infer(H));
+        expr arg = m_ctx.relaxed_whnf(m_ctx.infer(a));
+        expr pi, lhs, rhs;
```

Comment:
`arg` is not use. Moreover, this line may produce timeouts (Example: `arg` is `10000 + 10000`).